frontend/packages/ceph-storage-plugin/src/components/namespace-store/create-ns.tsx

```diff
@@ -0,0 +1,251 @@
+    }
+
+    promises.push(k8sCreate(NooBaaNamespaceStoreModel, nsPayload));
+    return handlePromise(Promise.all(promises), (resource) => {
+      const lastIndex = resource.length - 1;
+      if (isPage)
+        history.push(
+          `/k8s/ns/${namespace}/clusterserviceversions/${getName(csv)}/${referenceForModel(
+            NooBaaNamespaceStoreModel,
+          )}/${getName(resource[lastIndex])}`,
+        );
+      else close();
+    });
+  };
+
+  return (
+    <Form className={classNames('nb-bs-ns-form', className)} onSubmit={onSubmit} noValidate={false}>
+      <FormGroup
+        label={t('ceph-storage-plugin~Namespace Store Name')}
+        fieldId="namespacestore-name"
+        className="nb-bs-ns-form-entry"
+        helperText={t(
+          'ceph-storage-plugin~A unique name for the namespace store within the project',
+        )}
+        isRequired
+      >
+        <Tooltip
+          content="Name can contain a max of 43 characters"
+          isVisible={nsName.length > 42}
+          trigger="manual"
+        >
+          <TextInput
+            id="ns-name"
+            onChange={handleNsNameTextInputChange}
+            value={nsName}
+            maxLength={43}
+            data-test="namespacestore-name"
+            placeholder="my-namespacestore"
+            aria-label={t('ceph-storage-plugin~Namespace Store Name')}
+          />
+        </Tooltip>
+      </FormGroup>
+
+      <FormGroup
+        label={t('ceph-storage-plugin~Provider')}
+        fieldId="provider-name"
+        className="nb-bs-ns-form-entry"
+        isRequired
+      >
+        <Dropdown
+          id="providers"
+          className="nb-bs-ns-form-entry__dropdown"
+          buttonClassName="nb-bs-ns-form-entry__dropdown"
+          dataTest="namespacestore-provider"
+          onChange={setProvider}
+          items={PROVIDERS}
+          selectedKey={provider}
+        />
+      </FormGroup>
+      {(provider === BC_PROVIDERS.AWS ||
+        provider === BC_PROVIDERS.S3 ||
+        provider === BC_PROVIDERS.IBM ||
+        provider === BC_PROVIDERS.AZURE) && (
+        <S3EndPointType
+          type={StoreType.NS}
+          provider={provider}
+          namespace={CEPH_STORAGE_NAMESPACE}
+          state={providerDataState}
+          dispatch={providerDataDispatch}
+        />
+      )}
+      <ButtonBar errorMessage={errorMessage} inProgress={inProgress}>
+        <ActionGroup>
+          <Button type="submit" data-test="namespacestore-create-button" variant="primary">
+            {t('ceph-storage-plugin~Create')}
+          </Button>
+          <Button onClick={cancel} variant="secondary">
+            {t('ceph-storage-plugin~Cancel')}
+          </Button>
+        </ActionGroup>
+      </ButtonBar>
+    </Form>
+  );
+});
+
+export default CreateNamespaceStoreForm;
+
+type CreateNamespaceStoreFormProps = ModalComponentProps & {
+  isPage?: boolean;
+  namespace?: string;
```

Comment:
I believe you need namespace always - creating secret when its not selected
```suggestion
  namespace: string;
```

Reply:
changed